remoter-server/src/cellSizeDistribution.r

```diff
@@ -0,0 +1,99 @@
+                                        group.column = "orig.ident",
+                                        # [HARDCODED]
+                                        threshold.low = 1e2,
+                                        threshold.high = NULL
+  )
+  # extracting the inflection point value which can serve as minCellSize threshold
+  # all other side effects to the scdate object will be discarded
+  # [TODO unittest]: this calculation assumes a single sample, i.e. only one group in orig.ident!
+  # otherwise it will return multiple values for each group!
+  # returned is both the rank(s) as well as inflection point
+  # orig.ident          nCount_RNA  rank
+  # SeuratProject       1106        10722
+  tmp <- Tool(seurat_obj_tmp, slot = "CalculateBarcodeInflections")$inflection_points
+  # extracting only inflection point(s)
+  return(tmp$nCount_RNA)
+}
+
+task <- function(seurat_obj, config) {
+    import::here(map2, .from = purrr)
+    minCellSize <- as.numeric(config$filterSettings["minCellSize"])
+    # extract plotting data of original data to return to plot slot later
+    plot1_data  <- seurat_obj$nCount_RNA
+    # plot 1: histgram of UMIs, hence input is all UMI values, e.g.
+    # AAACCCAAGCGCCCAT-1 AAACCCAAGGTTCCGC-1 AAACCCACAGAGTTGG-1
+    #               2204              20090               5884  ...
+    #names(plot1_data) <- rep("u", length(plot1_data))
+   
+    #    u    u    u    u    u    u
+    # 3483 6019 3892 3729 4734 3244
+    plot2_data <- order(seurat_obj$nCount_RNA)
+    #names(plot2_data) <- rep("rank", length(plot2_data))
+    plot2_data <- unname(map2(plot1_data,plot2_data,function(x,y){c("u"=x,"rank"=y)}))
+    plot1_data <- lapply(unname(plot1_data),function(x) {c("u"=x)})
+    # Check if it is required to compute sensible values. From the function 'generate_default_values_cellSizeDistribution', it is expected
+    # to get a list with two elements {minCellSize and binStep}
+    if (as.logical(toupper(config$auto)))
+        # config not really needed for this one (maybe later for threshold.low/high):
+        minCellSize <- generate_default_values_cellSizeDistribution(seurat_obj, config)
+    # Check wheter the filter is set to true or false
+    # if true overwrite seurat object (side effect!)
+    if (as.logical(toupper(config$enabled))) {
+        # Information regarding number of UMIs per cells is pre-computed during the 'CreateSeuratObject' function. 
+        seurat_obj <- subset(seurat_obj, subset = nCount_RNA >= minCellSize)
+    }
+    # update config
+    config$filterSettings$minCellSize <- minCellSize
+    # the result object will have to conform to this format: {data, config, plotData : {plot1, plot2}}
+    result <- list(
+        data = seurat_obj,
+        config = config,
+        plotData = list(
+            cellSizeDistributionHistogram = plot1_data,
+            # Q: are both plots updated for this filter?
+            # Q: what is the format of plot2?
```

Comment:
you can see the format of plot2 in here: https://docs.google.com/document/d/1pEhYSzAYiYqQp8wPTPjQMGH3Nk8CvCrQpnVYW3exwYE/edit#heading=h.fxnlmdhqen2z

Reply:
This were some comments leftover from before, I just didn't delete them in case some of this questions are still relevant, but I'll delete them I think because they are already on data ingest.